### PR TITLE
Decoupling geometry and bc

### PR DESCRIPTION
**Pull request by finsberg**
Quite a lot of changes here, the main one being that you now need to send in a path to a geometry rather than creating it on the fly. For this we will use another package that I just created called [cardiac-geometries](https://computationalphysiology.github.io/cardiac_geometries/).

Apart for this I have also refactored the boundary conditions in a separate module. I the coming week I will try to add a demo for a left ventricle in which case we need completely different boundary conditions for the mechanics and so the logic for this should be implemented in this new module. 

I also removed some stuff that were saved in the Runner. We can try to make this as part of the datacollector in stead

### CODE REVIEW

File: src/simcardems/cli.py [hunk 1]
```diff
@@ -23,76 +22,67 @@ def cli():
...
-    default=Config.lz,
-    type=float,
-    help="Size of mesh in z-direction",
-)
 @click.option(
     "--bnd_cond",
-    default=Config.bnd_cond,
-    type=click.Choice(mechanics_model.BoundaryConditions._member_names_),
+    default=config.Config.bnd_cond,
+    type=click.Choice(config.SlabBoundaryConditionTypes._member_names_),
```
Comment (IlsevanHerck) [inline src/simcardems/cli.py:73]: Do I understand it correctly that these options are only available for the slab? If yes, we should add that in the description.

**Review by IlsevanHerck — APPROVED** [review]
There are a lot of changes, but it looks good to me. 
I made a small change to the demos to point the path for the geometry to the correct folder.

File: demos/release_test.py [hunk 2]
```diff
@@ -86,8 +86,8 @@ def postprocess(outdir: Path):
 
 if __name__ == "__main__":
     outdir = Path("release_test_results")
-    geometry_path = "geometries/slab.h5"
```
Comment (IlsevanHerck) [inline demos/release_test.py]: I just realised that this path depends on the local path from where the demo is started. Can we make this dependent on the path where the demo is located, not the path from where the demo is started?

Comment (finsberg) [reply demos/release_test.py]: Yes we can. Just did it here: https://github.com/ComputationalPhysiology/simcardems/pull/102/commits/8124f7137f29caf0c215b1ccdb2c7cacbe955cdf

File: demos/simple_demo.py [hunk 3]
```diff
@@ -13,31 +13,38 @@
 
 # Create configurations with custom output directory
 outdir = Path("results_simple_demo")
-config = simcardems.Config(outdir=outdir)
+
+# Specify paths to the geometry that we will use
+geometry_path = "geometries/slab.h5"
+geometry_schema_path = "geometries/slab.json"
```
[the remark this answers is not in the document]
Comment (finsberg) [reply demos/simple_demo.py]: Whoops, they will be added in the PR into this branch

File: src/simcardems/cli.py (hunk 1 above)
Comment (finsberg) [reply src/simcardems/cli.py:73]: Well, the only place this is used it here: https://github.com/ComputationalPhysiology/simcardems/blob/171e87a572de349a20bc6c4295f07fc8759d6611/src/simcardems/mechanics_model.py#L254 
Meaning that if it is `rigid` then it has to be a slab, so this is probably a bad name.

Comment (IlsevanHerck) [reply src/simcardems/cli.py]: I changed the boundary condition to a flag if the rigid motion boundary conditions should be used. See commit [[dad193e](https://github.com/ComputationalPhysiology/simcardems/pull/97/commits/dad193e5a75b6492f3b8e85005278c20eaaf30e2)](url)

**Pull request merged**